src/db/gc/mod.rs

```diff
@@ -163,7 +163,10 @@ impl<DB: Blockstore + GarbageCollectable + Sync + Send + 'static> MarkAndSweep<D
     /// using CAR-backed storage with a snapshot, for implementation simplicity.
```

Comment:
Is this note still valid, if not please update it.
```rust
/// NOTE: This currently does not take into account the fact that we might be starting the node
/// using CAR-backed storage with a snapshot, for implementation simplicity.
```

Reply:
This not will always be valid, it states that it does not care about car-backed storage blocks.

Reply:
In this case, I don't quite understand why `fn filter` works when it's takes the parity-db only. Shouldn't `fn unordered_stream_graph` take the entire DB (parity-db + car-db)?

Reply:
It should work just fine. We only care about cleaning up unreachable nodes. Since all the data that's stored in ParityDB is in the future for any data that's in car-db - we don't really care about it. Unless I'm missing some weird edge-case. I'm assuming that snapshot we export are finalized, in which case there should not be any issues.

The one enhancement I think we should do based on your comments and my revisiting the code is specifying the first epoch that's stored in the database as opposed to the car file. Otherwise
```rust
        if depth > current_epoch {
            time::sleep(interval).await;
            return anyhow::Ok(());
        }

````
is basically useless. Or we can get rid of it altogether and just rely on initial sleep that we do later for simplicity.

Reply:
> I'm assuming that snapshot we export are finalized

I'm not positive about this, as I understand this can only be true when we export snapshot from `latest - finality`, @lemmih could you weight in on this?

> just rely on initial sleep 

I suggest storing the last GC epoch in parity-db metadata comlumn, instead of waiting for 10h on every start. In an edge case, if we have a node that restarts (maybe for getting updates) every 9h then GC would never get a chance to run.

Reply:
@lemmih what do you think about the above? Seems reasonable to me. 
We could also initialize the last GC epoch as the heaviest tipset we see in car-backed storage when the database is empty.